aws-android-sdk-appsync-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java

```diff
@@ -90,27 +90,28 @@ public void interceptAsync(@Nonnull final InterceptorRequest request, @Nonnull f
       @Override public void run() {
         callBack.onFetch(FetchSourceType.NETWORK);
 
-          try {
+        try {
           httpCall = httpCall(request.operation);
         } catch (IOException e) {
           logger.e(e, "Failed to prepare http call for operation %s", request.operation.name().name());
           callBack.onFailure(new ApolloNetworkException("Failed to prepare http call", e));
           return;
         }
-
-        httpCall.enqueue(new Callback() {
-          @Override public void onFailure(@Nonnull Call call, @Nonnull IOException e) {
-            if (disposed) return;
-            logger.e(e, "Failed to execute http call for operation %s", request.operation.name().name());
-            callBack.onFailure(new ApolloNetworkException("Failed to execute http call", e));
-          }
-
-          @Override public void onResponse(@Nonnull Call call, @Nonnull Response response) throws IOException {
-            if (disposed) return;
-            callBack.onResponse(new ApolloInterceptor.InterceptorResponse(response));
-            callBack.onCompleted();
-          }
-        });
+        if (httpCall != null) {
```

Comment:
Instead of a silent null check can it run the onFailure callback with `ApolloNetworkException("Failed to prepare http call, prepared call was null")` and return?

Reply:
I have amended the callback to return the ApolloNetworkException. Please see commit - 4a2cb0b3848484f88175575c72d6e87fc8525189. Appreciate if you could release this on your side if you are agreeable to this change.